packages/flutter/lib/src/material/tooltip.dart

```diff
@@ -755,13 +758,21 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
       verticalOffset: widget.verticalOffset ?? tooltipTheme.verticalOffset ?? _defaultVerticalOffset,
       preferBelow: widget.preferBelow ?? tooltipTheme.preferBelow ?? _defaultPreferBelow,
     );
+
+    return SelectionContainer.maybeOf(context) == null
+      ? overlayChild
+      : SelectionContainer.disabled(child: overlayChild);
   }
 
   @override
   void dispose() {
     GestureBinding.instance.pointerRouter.removeGlobalRoute(_handleGlobalPointerEvent);
     Tooltip._openedTooltips.remove(this);
+    // Remove the onCancel callbacks to prevent the registered callbacks from
+    // triggering unnecessary sideeffects (such as animation).
+    _longPressRecognizer?.onLongPressCancel = null;
```

Comment:
This is surprising, why would it still fire if `_longPressRecognizer` is disposed?

Reply:
This seems weird to me, the on cancel usually used for cleaning up the state in the owner when gesture is in progress. The owner won't expect the on cancel to be called if it is actively disposing the recognizer. it would be the owner's responsibility to clean up and state left behind when disposing the resource. WDYT?

Reply:
From the gesture recognizer's perspective I guess that's expected? When there's an `onGestureStart`  someone would expect either an `onGestureUp` or `onGestureCancel`. The owner of the gesture recognizer may not be the owner of the gesture callbacks ?

e.g. `SomeWidgetWhoseElementOwnsALongPressGestureRecognizer(this.onLongPress, this.onLongPressCancel)`, when this widget unmounts the parent widget may appreciate that they're getting an `onLongPressCancel` to pair up with the `onLongPress` they received previously?